packages/affiliates/libs/affiliates.js

```diff
@@ -144,32 +147,49 @@ const DeployerRewards = ({ queries, empCreatorAbi, empAbi, coingecko, synthPrice
     }, []);
 
     // Per snapshot calculate the associated amount that each deployer is entitled to.
-    let finalDeployerPayouts = valuePerSnapshot.reduce((result, valueByEmp) => {
-      const totalValueLocked = valueByEmp.reduce((result, [, value]) => {
-        return result.add(toBN(value));
-      }, toBN("0"));
-      valueByEmp.forEach(([emp, value]) => {
-        const deployer = empDeployers.get(emp);
-        const contribution =
-          totalValueLocked.toString() != "0"
-            ? toBN(value) // eslint-disable-line indent
-                .mul(toBN(toWei("1"))) // eslint-disable-line indent
-                .div(totalValueLocked) // eslint-disable-line indent
-            : toBN("0"); // eslint-disable-line indent
-        const rewards = contribution.mul(payoutPerSnapshot).div(toBN(toWei("1")));
-        if (result[deployer] == null) result[deployer] = toBN("0");
-        result[deployer] = result[deployer].add(rewards);
-      });
+    let finalPayouts = valuePerSnapshot.reduce(
+      ({ deployerPayouts, empPayouts }, valueByEmp) => {
+        const totalValueLocked = valueByEmp.reduce((result, [, value]) => {
+          return result.add(toBN(value));
+        }, toBN("0"));
+        valueByEmp.forEach(([emp, value]) => {
+          const deployer = empDeployers.get(emp);
+          const contribution =
+            totalValueLocked.toString() != "0"
+              ? toBN(value) // eslint-disable-line indent
+                  .mul(toBN(toWei("1"))) // eslint-disable-line indent
+                  .div(totalValueLocked) // eslint-disable-line indent
+              : toBN("0"); // eslint-disable-line indent
+          const rewards = contribution.mul(payoutPerSnapshot).div(toBN(toWei("1")));
+
+          // calculate deployer rewards
+          if (deployerPayouts[deployer] == null) deployerPayouts[deployer] = toBN("0");
+          deployerPayouts[deployer] = deployerPayouts[deployer].add(rewards);
+
+          // calculate per emp rewards
+          if (empPayouts[emp] == null) empPayouts[emp] = toBN("0");
+          empPayouts[emp] = empPayouts[emp].add(rewards);
+        });
 
-      return result;
-    }, {});
+        return { deployerPayouts, empPayouts };
+      },
+      { deployerPayouts: {}, empPayouts: {} }
```

Comment:
this just adds a counters for deployer and emppayouts and seperate objects. None of the math has changed